libcxi: add new versions, remove version build requirement

jscook2345 opened the pull request:
This adds several new versions from git tags and removed the `@main` version requirement from building.

jscook2345 committed:
packages: libcxi: add new versions using git tags

https://github.com/HewlettPackard/shs-libcxi/tags

Signed-off-by: Justin Cook <jscook@lbl.gov>

jscook2345 committed:
packages: libcxi: remove build requirement of version being main

Signed-off-by: Justin Cook <jscook@lbl.gov>

rbberger reviewed (changes requested):
You also need to remove this:

https://github.com/jscook2345/spack-packages/blob/1c533d6652bd220c1c7c3a6fce3559c4eac1dfec/repos/spack_repo/builtin/packages/libcxi/package.py#L62

jscook2345 commented:
> You also need to remove this:
> 
> https://github.com/jscook2345/spack-packages/blob/1c533d6652bd220c1c7c3a6fce3559c4eac1dfec/repos/spack_repo/builtin/packages/libcxi/package.py#L62

Thanks for spotting this! Fixed, rebased, and pushed!
